src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php

```diff
@@ -331,4 +340,32 @@ protected function createRoute(string $path, array $defaults, array $requirement
     }
 
     abstract protected function configureRoute(Route $route, \ReflectionClass $class, \ReflectionMethod $method, $annot);
+
+    /**
+     * @param \ReflectionClass|\ReflectionMethod $reflection
+     *
+     * @return iterable<RouteAnnotation>
+     */
+    private function getAnnotations(object $reflection): iterable
+    {
+        if (method_exists($reflection, 'getAttributes')) {
+            foreach ($reflection->getAttributes($this->routeAttributeClass) as $attribute) {
+                yield $attribute->newInstance();
```

Comment:
isn't it dangerous to use a `newInstance` ? or are the args taken from the attribute itself (didn't look that part)

Reply:
ehh nevermind, saw the tests

Reply:
Coming from Doctrine annotations, this looks odd, I admit. But the reflection API does not instantiate the actual attribute unless you tell it to.